server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -711,15 +710,11 @@ public NicSecondaryIp allocateSecondaryGuestIP(final long nicId, IpAddresses req
         String ip6addr = null;
         //Isolated network can exist in Basic zone only, so no need to verify the zone type
         if (network.getGuestType() == Network.GuestType.Isolated) {
-            try {
-                if (ipv4Address != null) {
-                    ipaddr = _ipAddrMgr.allocateGuestIP(network, ipv4Address);
-                }
-                if (ipv6Address != null) {
-                    ip6addr = ipv6AddrMgr.allocateGuestIpv6(network, ipv6Address);
-                }
-            } catch (InsufficientAddressCapacityException e) {
-                throw new InvalidParameterValueException("Allocating guest ip for nic failed");
+            if ((ipv4Address != null || NetUtils.isIpv4(network.getGateway()) && org.apache.commons.lang3.StringUtils.isBlank(ipv6Address))) {
```

Comment:
This would imply that ipv4 address is allocated only when ipv6 is not passed. Do you think this may cause issue for ipv6 users /cc @wido ?

Reply:
No, it would not. The user can call the method to ask an IPv4 (is the network supports). And then, in a second call, he/she can ask for an IPv6 as well. The only difference now is that for IPv6 an IP must be defined by the user. ACS will not generate one.

Reply:
That is the idea indeed @rafaelweingartner @rhtyd.
It was designed to allow allocating ipv4 or ipv6 (does not accept null ipv6). If a user wants to allocate both, IPv4 and IPv6, he/she needs to call the API one time per IP address.
My mistake here was adding a null verification for ipv4 addresses as well.